plugins/in_stdin/in_stdin.c

```diff
@@ -210,45 +211,97 @@ static int in_stdin_collect(struct flb_input_instance *ins,
     return 0;
 }
 
+/* Read stdin config*/
+static int in_stdin_config_init(struct flb_in_stdin_config *ctx,
+                               struct flb_input_instance *in,
+                               struct flb_config *config)
+{
+    const char *pval = NULL;
+
+    ctx->buf_size = DEFAULT_BUF_SIZE;
+    ctx->buf = NULL;
+    ctx->buf_len = 0;
+    ctx->ins = in;
+
+    /* parser settings */
+    pval = flb_input_get_property("parser", in);
+    if (pval) {
+        ctx->parser = flb_parser_get(pval, config);
+        if (!ctx->parser) {
+            flb_plg_error(ctx->ins, "requested parser '%s' not found", pval);
+            return -1;
+        }
+    }
+
+    /* buffer size setting */
+    pval = flb_input_get_property("buffer_size", in);
+    if (pval != NULL) {
+        ctx->buf_size = (size_t) flb_utils_size_to_bytes(pval);
+
+        if (ctx->buf_size == -1) {
+            flb_plg_error(ctx->ins, "buffer_size '%s' is invalid", pval);
+            return -1;
+        }
+        else if (ctx->buf_size < DEFAULT_BUF_SIZE) {
+            flb_plg_error(ctx->ins, "buffer_size '%s' must be at least %i bytes",
+                          pval, DEFAULT_BUF_SIZE);
+            return -1;
+        }
```

Comment:
@sxd any idea how to do this better ?

Reply:
@dojci actually I think it's really easy to understand that piece of code, in my opinion it's fine and I'm ok with it